projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -282,6 +288,33 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseDirecti
         return !!this.childLayoutKeys.length;
     }
 
+    /**
+     * Gets/Sets row selection mode
+     *
+     * @remarks
+     * By default the row selection mode is none
+     * @param selectionMode: FlatGridSelectionMode
+     */
+    @Input()
+    get rowSelection() {
+        return this._rowSelectionMode;
+    }
+
+    set rowSelection(selectionMode: GridSelectionMode) {
+        this._rowSelectionMode = selectionMode;
+        if (!this._init) {
+            this.selectionService.clearAllSelectedRows();
+            this.notifyChanges(true);
+        }
+    }
+
+    /**
+     * @hidden @internal
+     */
+    public get isMultiRowSelectionEnabled(): boolean {
```

Comment:
You don't need this either. Leave the base implementation take care of it. It will return true on GridSelectionMode.multiple and on GridSelectionMode.multipleCascade